### PR TITLE
Dark mode

### PR DESCRIPTION
**Pull request by oskarrough**
Added a button that toggles theme and some basic black/white styles.

### CODE REVIEW

File: src/components/theme-toggle-button.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
+// Needed for the CSS.
+import icon from './theme-toggle-button.png'
```
Comment (hugurp) [inline src/components/theme-toggle-button.js:2]: D: !!!!!

File: src/components/theme-toggle-button.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+const DARK = '(prefers-color-scheme: dark)'
+const DARKCLASS = 'dark-theme'
+
+export default function ThemeToggleButton() {
+	// The <html> element.
+	const el = document.documentElement
+
+	// Set class if user prefers darkness.
+	if (window.matchMedia(DARK).matches) {
+		el.classList.add(DARKCLASS)
```
Comment (hugurp) [inline src/components/theme-toggle-button.js:13]: think react would like us no to manipulate classNames like this anywhere in the app.
 One thing we can use, would be close to an `ember.service`, a `react.context`.
 https://reactjs.org/docs/context.html
 That is, if we want anything to do with, for example a `UserStyles` (?) context, which could give some data to lower components, without having to pass all the data down through each parent/child (by connecting to the `react.context.user_styles`

File: src/index.css [hunk 3]
```diff
@@ -12,3 +38,46 @@ body {
...
+	opacity: 0.8;
+}
+.ThemeToggleButton:focus {
+	outline: none;
+}
+.ThemeToggleButton:hover {
+	opacity: 1;
+	cursor: pointer;
+}
+.dark-theme .ThemeToggleButton {
```
Comment (hugurp) [inline src/index.css:81]: conflicting css/html naming convention?
what do we go for in a react context?

File: src/components/theme-toggle-button.js (hunk 1 above)
Comment (oskarrough) [reply src/components/theme-toggle-button.js:2]: 🤦

File: src/index.css (hunk 3 above)
Comment (oskarrough) [reply src/index.css:81]: for the `.dark-theme`? Yea not sure. Utility? So `.is-dark` maybe?

File: src/components/theme-toggle-button.js (hunk 2 above)
Comment (oskarrough) [reply src/components/theme-toggle-button.js:13]: True makes sense.

Right now it relies on the `<html>` element. Could it target the top level react element instead?

Comment (hugurp) [reply src/components/theme-toggle-button.js:13]: Yes! The link to the context doc is implementing dark mode as an example :}

File: src/index.css (hunk 3 above)
Comment (hugurp) [reply src/index.css:81]: no clue!
I'd say a classic BEM? with emphasis on components, and matching the naming of css/html to the naming of react?

File: src/components/theme-toggle-button.js (hunk 2 above)
Comment (oskarrough) [reply src/components/theme-toggle-button.js:13]: Haha, the guide even has a theme toggle button!

We can try that if/when we need access to the "theme" inside javascript land? At the moment it seems fine with just CSS land.

File: src/index.css (hunk 3 above)
Comment (oskarrough) [reply src/index.css:81]: i only know the suit css ones from back in the day but I follow you here!

File: src/Layout.js [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+	const darkModeListener = (event) => {
+		if (event.matches) {
+			setTheme(themeContextDefault.themes[0])
+		} else {
+			setTheme(themeContextDefault.themes[1])
+		}
+	}
+
+	useEffect(() => {
+		window.addEventListener('change', darkModeListener)
```
Comment (hugurp) [inline src/Layout.js:32]: not sure why this is never called

Comment (oskarrough) [reply src/Layout.js:32]: the evenlistener should not be on `window` but `window.matchMedia`. Like this:

https://github.com/internet4000/radio4000-cms/blob/a31ca31579e90b2bb3a235cfefe72cb7322fb388/src/contexts/theme.js#L13

**Pull request merged**